tests/functional/event_handler/required_dependencies/appsync/test_appsync_batch_resolvers.py

```diff
@@ -942,3 +649,181 @@ async def get_locations2(event: AppSyncResolverEvent, name: str) -> str:
     # THEN Verify that the results match the expected values
     assert result1 == ["get_locations#value#1"]
     assert result2 == ["get_locations2#value#2"]
+
+
+def test_resolve_batch_processing_with_simple_queries_with_aggregate():
+    # GIVEN a list of events representing GraphQL queries for listing locations
+    event = [
+        {
+            "typeName": "Query",
+            "info": {
+                "fieldName": "listLocations",
+                "parentTypeName": "Post",
+            },
+            "fieldName": "listLocations",
+            "arguments": {},
+            "source": {
+                "id": "1",
+            },
+        },
+        {
+            "typeName": "Query",
+            "info": {
+                "fieldName": "listLocations",
+                "parentTypeName": "Post",
+            },
+            "fieldName": "listLocations",
+            "arguments": {},
+            "source": {
+                "id": "2",
+            },
+        },
+        {
+            "typeName": "Query",
+            "info": {
+                "fieldName": "listLocations",
+                "parentTypeName": "Post",
+            },
+            "fieldName": "listLocations",
+            "arguments": {},
+            "source": {
+                "id": [3, 4],
+            },
+        },
+    ]
+
+    app = AppSyncResolver()
+
+    # WHEN the batch resolver for the listLocations field is defined
+    @app.batch_resolver(field_name="listLocations")
+    def create_something(event: List[AppSyncResolverEvent]) -> Optional[list]:  # noqa AA03 VNE003
```

Comment:
I think you meant `list` instead of `Optional[list]`, otherwise this would fail in your exception.

I think you also have a test for that type of scenario too

Reply:
Yeah! Done